src/servers/src/grpc/flight/stream.rs

```diff
@@ -83,6 +83,11 @@ impl FlightRecordBatchStream {
                 }
             }
         }
+        // make last package to pass metrics
+        if let Some(m) = recordbatches.metrics() {
+            let metrics = FlightMessage::Metrics(m);
+            let _ = tx.send(Ok(metrics)).await;
+        }
```

Comment:
cc @waynexia 
If transmission of metrics fails, I guess that it might be better to inform `frontend` about it. This might be because `frontend` continue to wait for metrics corresponding to each distributed query when executing `DISTRIBUTED ANALYZE PLAN`

Reply:
What do you mean by "inform"? The current implementation puts metrics at the end of each stream. This is because the metric is only available when a corresponding plan is finished.

For distributed analyze scenario, what do you think of discarding data and making the metric the first and only result?